### ISSUE TITLE
[HOLD for payment 2025-01-02] [$125]  iOS Navigation - Inbox, Search and Settings open in slow-motion

### ISSUE THREAD

**Issue by IuliiaHerets** [Expensify/App#53347]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 9.0.69-1
**Reproducible in staging?:** Y
**Reproducible in production?:** N
**If this was caught on HybridApp, is this reproducible on New Expensify Standalone?:** Y
**Email or phone of affected tester (no customers):** applausetester+232803@applause.expensifail.com
**Issue reported by:** Applause Internal Team

## Action Performed:
1. Launch ND or hybrid app.
2. Navigate between Inbox, Search and Settings.

## Expected Result:
Inbox, Search and Settings will open without slow-motion.

## Actual Result:
Inbox, Search and Settings open in slow-motion.

## Workaround:
Unknown

## Platforms:

- [ ] Android: Standalone
- [ ] Android: HybridApp
- [ ] Android: mWeb Chrome
- [x] iOS: Standalone
- [x] iOS: HybridApp
- [ ] iOS: mWeb Safari
- [ ] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos
  
https://github.com/user-attachments/assets/edd5533d-462e-4ab4-b47a-9ef1f3b9f783

</details>

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~021865056809629169420</li>
        <li>Upwork Job ID: 1865056809629169420</li>
        <li>Last Price Increase: 2024-12-06</li>
    </ul>
</details>

<details><summary>Issue Owner</summary>Current Issue Owner: @johncschuster</details>

**Comment by mountiny** [comment]
This was a change by design as we switched to native stack and i dont think its blocking deploy. 

I believe design team did not have any objections but could you please weigh in again (it has been a while it has been discussed on the pr) cc @Expensify/design

**Comment by dubielzyk-expensify** [comment]
I can't test this as I can't pull this version yet, but I don't mind the fade between the screens. I'd personally speed up the fade animation a tad cause when navigating quickly between the tabs it can often cause issues or feel slow if it's not snappy enough. I'm happy to do that as a follow-up bug though.

Keen to hear what rest of @Expensify/design thinks here

**Comment by mountiny** [comment]
I am +1 to making this making twice the speed

**Comment by trjExpensify** [comment]
Same, agree with you both. 👍

**Comment by dannymcclain** [comment]
Agree with all of the above, let's send it!

**Comment by dubielzyk-expensify** [comment]
Would also be good if we could somehow test this on some slower devices and slower internet just to make sure that it still works well. 

I'm testing this in prod on my iPhone 15 Pro and while speeding up the animation will help, there also seem to be a massive delay when I tap the bottom nav to when it turns to active state. Even before the animation kicks in.

**Comment by mountiny** [comment]
> there also seem to be a massive delay when I tap the bottom nav to when it turns to active state

Can you show a video? I dont see any significant delay on my side in ios

**Comment by dubielzyk-expensify** [comment]
Compared to other apps it seems quite a bit slower and less responsive:

https://github.com/user-attachments/assets/0157e30a-cc7c-4df4-8091-53bfa9cc5465

**Comment by mountiny** [comment]
@chrispader @kirillzyusko, could you please prepare a PR with half the animation time, and we could iterate on this? I assume the animation just adds up to the slower feeling.

**Comment by dannymcclain** [comment]
>Compared to other apps it seems quite a bit slower and less responsive:

I just got this version on my phone yesterday and it does feel slower/less responsive to me too. But I don't think it's just the screen animation (which I agree we should speed up)—you can see in Jon's video that there's a delay in the _tab highlight_ after he presses as well. IMO it is ok if the bottom tab highlight and screen animation are not in sync. I think the tab should highlight _immediately_ upon pressing to give the user that feedback.

**Comment by dubielzyk-expensify** [comment]
Exactly. I'm happy to start with speeding up, but it's pretty clear that there's a delay from the tap before the animation triggers

**Comment by mountiny** [comment]
Making this external as it can be handled by contributors. Can you please speed up the animation by switching between the bottom tabs to twice the speed? Thanks!

**Comment by kirillzyusko** [comment]
Hey, I'm Kiryl from Margelo expert agency and I'd like to work on this issue 👋 

> I think the tab should highlight immediately upon pressing to give the user that feedback.

You are right, but I think it's only `Search` tab that doesn't switch immediately, right? Inbox/Settings switch happens almost immediately. When I debugged `Search` page it had some slow re-renders, so I think it's causing this 😔

**Comment by mountiny** [comment]
We could look into the re-renders separately for sure

**Comment by kirillzyusko** [comment]
> We could look into the re-renders separately for sure

Yes, it would be better to create a separate issue for that 🙏

**Comment by mountiny** [comment]
Asked here https://expensify.slack.com/archives/C05LX9D6E07/p1733757346965449

**Comment by Kicu** [comment]
FYI @kirillzyusko take a look at the most recent PR from me: https://github.com/Expensify/App/pull/53198 I hoped it would speed things up a bit, I had to move a bunch of code to another component, but tried to wrap everything I can in useMemo/useCallback.

**Comment by zirgulis** [comment]
@mountiny I've create a new task to investigate slow rerenders https://github.com/Expensify/App/issues/53818

**Comment by mvtglobally** [comment]
Issue not reproducible during KI retests. (First week)

**Comment by johncschuster** [comment]
@DylanDylann, I noticed in the above comment that you're supposed to request payment via NewDot, but I don't see you listed on my end as a team member that can receive payments through ND yet. Can you clarify that for me so I can make sure you're paid correctly?

**Comment by DylanDylann** [comment]
I still get paid via Upwork

**Comment by DylanDylann** [comment]
This is an improvement with label NoQA on PR so I believe that we don't need checklist

**Comment by johncschuster** [comment]
This is paid! Thanks for your contributions!
